#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kuka/youbot/protos/BodyMesh.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kuka/youbot/protos/InteriorWheel.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kuka/youbot/protos/ExteriorWheel.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/Parquetry.proto"

WorldInfo {
  info [
    "Simulation of the KUKA youBot robot."
    "This model includes an accurate and fast simulation of the Meccanum wheels."
    "The robot can be equiped with various sensors (including the Microsoft Kinect) and configured with zero, one or two arms."
  ]
  title "KUKA youBot"
  basicTimeStep 4
  contactProperties [
    ContactProperties {
      material1 "InteriorWheelMat"
      coulombFriction [
        0, 2, 0
      ]
      frictionRotation -0.785398 0
      bounce 0
      forceDependentSlip [
        10, 0
      ]
    }
    ContactProperties {
      material1 "ExteriorWheelMat"
      coulombFriction [
        0, 2, 0
      ]
      frictionRotation 0.785398 0
      bounce 0
      forceDependentSlip [
        10, 0
      ]
    }
  ]
}
Viewpoint {
  orientation -0.5773502691896257 0.5773502691896257 0.5773502691896257 2.0943245915045035
  position -6.336962168390665e-05 -0.0001250060200211522 9.725555704782863
  follow "youBot"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF YOUBOT Robot {
  translation 0 0 0.102838
  children [
    Compass {
    }
    GPS {
    }
    InertialUnit {
      name "IMU"
    }
    BodyMesh {
    }
    Group {
    }
    DEF WHEEL1 InteriorWheel {
      translation 0.228 -0.158 -0.055
      anchor 0.228 -0.158 -0.055
      name "wheel1"
      sensorName "enc1"
    }
    DEF WHEEL2 ExteriorWheel {
      translation 0.228 0.158 -0.055
      anchor 0.228 0.158 -0.055
      name "wheel2"
      sensorName "enc2"
    }
    DEF WHEEL3 ExteriorWheel {
      translation -0.228 -0.158 -0.055
      anchor -0.228 -0.158 -0.055
      name "wheel3"
      sensorName "enc3"
    }
    DEF WHEEL4 InteriorWheel {
      translation -0.228 0.158 -0.055
      anchor -0.228 0.158 -0.055
      name "wheel4"
      sensorName "enc4"
    }
  ]
  name "youBot"
  model "KUKA youBot"
  description "KUKA youBot"
  boundingObject DEF BODY Group {
    children [
      Pose {
        translation 0 0 -0.045
        children [
          Box {
            size 0.34 0.34 0.09
          }
        ]
      }
      Pose {
        translation 0 0 -0.045
        children [
          Box {
            size 0.56 0.23 0.09
          }
        ]
      }
    ]
  }
  physics Physics {
    density -1
    mass 22
    centerOfMass [
      0 0 -0.045
    ]
    inertiaMatrix [
      0.166204 0.418086 0.55459
      0 0 0
    ]
  }
  controller "kuka_controller"
  supervisor TRUE
}
RectangleArena {
  floorSize 10 10
  floorTileSize 5 5
  floorAppearance Parquetry {
    type "light strip"
  }
}
DEF TARGET1 Solid {
  translation 3 0 0
  children [
    Shape {
      appearance PBRAppearance {
        metalness 0
      }
      geometry Box {
        size 0.7 0.5 0.001
      }
    }
  ]
}
DEF TARGET2 Solid {
  translation 3 3 0
  children [
    Shape {
      appearance PBRAppearance {
        metalness 0
      }
      geometry Box {
        size 0.7 0.5 0.001
      }
    }
  ]
  name "solid(1)"
}
DEF TARGET3 Solid {
  translation 0 -2 0
  children [
    Shape {
      appearance PBRAppearance {
        metalness 0
      }
      geometry Box {
        size 0.7 0.5 0.001
      }
    }
  ]
  name "solid(2)"
}
DEF TARGET4 Solid {
  translation -2 1 0
  children [
    Shape {
      appearance PBRAppearance {
        metalness 0
      }
      geometry Box {
        size 0.7 0.5 0.0001
      }
    }
  ]
  name "solid(3)"
}
